Stack/Opc.Ua.Core/Security/Certificates/CertificateIdentifier.cs

```diff
@@ -487,8 +503,187 @@ public ICertificateStore OpenStore()
             store.Open(this.StorePath, false);
             return store;
         }
+
+        /// <summary>
+        /// Retrieves the minimum accepted key size given the security configuration
+        /// </summary>
+        /// <param name="securityConfiguration"></param>
+        /// <returns></returns>
+        public ushort GetMinKeySize(SecurityConfiguration securityConfiguration)
+        {
+            if (CertificateType == ObjectTypeIds.EccBrainpoolP256r1ApplicationCertificateType ||
+                CertificateType == ObjectTypeIds.EccBrainpoolP384r1ApplicationCertificateType ||
+                CertificateType == ObjectTypeIds.EccCurve25519ApplicationCertificateType ||
+                CertificateType == ObjectTypeIds.EccCurve448ApplicationCertificateType ||
+                CertificateType == ObjectTypeIds.EccNistP256ApplicationCertificateType ||
+                CertificateType == ObjectTypeIds.EccNistP384ApplicationCertificateType)
+            {
+                return securityConfiguration.MinimumECCertificateKeySize;
+            }
+            else if (
+                CertificateType == ObjectTypeIds.RsaMinApplicationCertificateType ||
+                CertificateType == ObjectTypeIds.RsaSha256ApplicationCertificateType ||
+                securityConfiguration.IsDeprecatedConfiguration) // Deprecated configurations are implicitly RSA
+            {
+                return securityConfiguration.MinimumCertificateKeySize;
+            }
+            
+            throw new ArgumentException("Certificate type is unknown");
+            
+        }
+
+
+        /// <summary>
+        /// Get the OPC UA CertificateType.
+        /// </summary>
+        /// <param name="certificate">The certificate with a signature.</param>
+        public static NodeId GetCertificateType(X509Certificate2 certificate)
```

Comment:
Not shure about this comment: Shouldnt this function be part of X509Utils as well as the ValidateCertificateType function or should all OPC UA Specific logic be in CertificateIdentified